### PR TITLE
[feature](nereids) support match syntax

### PR DESCRIPTION
**Pull request by Tanya-W**
## Proposed changes

Support match syntax in nereids.
match syntax use like:
```sql
select * from test where msg match "hello";
select * from test where msg match_any "hello";
select * from test where msg match_all "hello hi";
select * from test where msg match_phrase "hello world";
```
`match` is same as `match_any`.

the pr of match syntax in original planner: https://github.com/apache/doris/pull/14211

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: fe/fe-core/src/main/antlr4/org/apache/doris/nereids/DorisParser.g4 [hunk 1]
```diff
@@ -613,6 +614,10 @@ nonReserved
     | LOGICAL
     | MACRO
     | MAP
+    | MATCH
+    | MATCH_ANY
+    | MATCH_ALL
+    | MATCH_PHRASE
```
Comment (morrySnow) [inline fe/fe-core/src/main/antlr4/org/apache/doris/nereids/DorisParser.g4:620]: in legacy planner, all match related keywords are reserved keywords, so remove them from nonReserved

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/visitor/ExpressionVisitor.java [hunk 2]
```diff
@@ -433,6 +434,10 @@ public R visitWindowFrame(WindowFrame windowFrame, C context) {
         return visit(windowFrame, context);
     }
 
+    public R visitMatch(Match match, C context) {
+        return visit(match, context);
+    }
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/visitor/ExpressionVisitor.java:439]: u should add all visitXXX here, not only add the base class. and all other visitXX should call `visitMatch`

**Comment by Tanya-W** [comment]
run buildall

File: regression-test/suites/nereids_syntax_p0/match.groovy [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+    """
+    
+    sql """ insert into test_nereids_match_select VALUES
+        ("zhang san", 10, "grade 5", "zhang yi", "chen san", "Class activists"),
+        ("zhang san yi", 11, "grade 5", "zhang yi", "chen san", "learn makes me happy"),
+        ("li si", 9, "grade 4", "li er", "wan jiu", "i just want go outside"),
+        ("san zhang", 10, "grade 5", "", "", ""),
+        ("li sisi", 11, "grade 6", "li ba", "li liuliu", "")
+    """
+
```
Comment (morrySnow) [inline regression-test/suites/nereids_syntax_p0/match.groovy:61]: add sync after insert

File: regression-test/suites/nereids_syntax_p0/match.groovy [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+    order_qt_match_6 """
+        SELECT fatherName, matherName FROM test_nereids_match_select WHERE name match 'zhang' or grade = 'grade 6';
+    """
+
+    order_qt_match_7 """
+        SELECT * FROM test_nereids_match_select WHERE name match 'zhang' and selfComment match 'happy';
+    """
+
```
Comment (morrySnow) [inline regression-test/suites/nereids_syntax_p0/match.groovy:90]: redundant blank line. if u want split different function test. the better way is add comment, not a redundant blank line

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/visitor/ExpressionVisitor.java (hunk 2 above)
Comment (Tanya-W) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/visitor/ExpressionVisitor.java:439]: I will update it

File: regression-test/suites/nereids_syntax_p0/match.groovy (hunk 4 above)
Comment (Tanya-W) [reply regression-test/suites/nereids_syntax_p0/match.groovy:90]: fixed

File: regression-test/suites/nereids_syntax_p0/match.groovy (hunk 3 above)
Comment (Tanya-W) [reply regression-test/suites/nereids_syntax_p0/match.groovy:61]: updated

File: fe/fe-core/src/main/antlr4/org/apache/doris/nereids/DorisParser.g4 (hunk 1 above)
Comment (Tanya-W) [reply fe/fe-core/src/main/antlr4/org/apache/doris/nereids/DorisParser.g4:620]: I will update it

**Comment by Tanya-W** [comment]
run buildall

**Review by morrySnow — APPROVED** [review]

**Pull request merged**